RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc

```diff
@@ -132,37 +132,38 @@ void ParticleNetSonicJetTagsProducer::acquire(edm::Event const &iEvent, edm::Eve
   client_->setBatchSize(tag_infos->size());
   skippedInference_ = false;
   if (!tag_infos->empty()) {
-    unsigned int maxParticles = 0;
-    unsigned int maxVertices = 0;
-    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
-      maxParticles = std::max(maxParticles,
-                              static_cast<unsigned int>(((*tag_infos)[jet_n]).features().get("pfcand_etarel").size()));
-      maxVertices =
-          std::max(maxVertices, static_cast<unsigned int>(((*tag_infos)[jet_n]).features().get("sv_etarel").size()));
-    }
-    if (maxParticles == 0 && maxVertices == 0) {
-      client_->setBatchSize(0);
-      skippedInference_ = true;
-      return;
-    }
     unsigned int minPartFromJSON = prep_info_map_.at(input_names_[0]).min_length;
     unsigned int maxPartFromJSON = prep_info_map_.at(input_names_[0]).max_length;
     unsigned int minVertFromJSON = prep_info_map_.at(input_names_[3]).min_length;
     unsigned int maxVertFromJSON = prep_info_map_.at(input_names_[3]).max_length;
-    maxParticles = std::clamp(maxParticles, minPartFromJSON, maxPartFromJSON);
-    maxVertices = std::clamp(maxVertices, minVertFromJSON, maxVertFromJSON);
+    bool toSkipInference = true;
 
     for (unsigned igroup = 0; igroup < input_names_.size(); ++igroup) {
       const auto &group_name = input_names_[igroup];
       auto &input = iInput.at(group_name);
-      unsigned target;
-      if (igroup < numParticleGroups_) {
-        input.setShape(1, maxParticles);
-        target = maxParticles;
-      } else {
-        input.setShape(1, maxVertices);
-        target = maxVertices;
+      for(unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n){
+        if (igroup == 0){
+          unsigned entry_target;
+          if (igroup < numParticleGroups_) {
+            entry_target = std::clamp(static_cast<unsigned int>(((*tag_infos)[jet_n]).features().get("pfcand_etarel").size()), minPartFromJSON, maxPartFromJSON);
+            if (toSkipInference && ((*tag_infos)[jet_n]).features().get("pfcand_etarel").size() > 0) toSkipInference = false;
+          } else {
+            entry_target = std::clamp(static_cast<unsigned int>(((*tag_infos)[jet_n]).features().get("sv_etarel").size()), minVertFromJSON, maxVertFromJSON);
+            if (toSkipInference && ((*tag_infos)[jet_n]).features().get("sv_etarel").size() > 0) toSkipInference = false;
+          }
+          input.setShape(1, entry_target, jet_n);
+        }
+        else{
+          input.setShape(1, static_cast<int64_t>(iInput.at(input_names_[igroup - 1]).shape(jet_n)[1]), jet_n);
+        }
       }
+
+      if (toSkipInference){
+        client_->setBatchSize(0);
+        skippedInference_ = true;
```

Comment:
we can remove `skippedInference_` everywhere - the new interface does not require it. (it now creates a fake, empty output vector when the batch size is set to 0.)

Reply:
Resolved (Commented it out for now)

Reply:
please remove it completely, including from the `produce()` function.

Reply:
Actually, from the `accquire()` function, `skippedInference_` is also set to true when there is an event with 0-sized jets i.e, when `tag_infos->empty()` is `false`, but all jets are 0-sized.
If we remove skippedInference_ from produce, then I will have to iterate over the feature jets again. I guess in this case `skippedInference_` is a neat way to remove an additional iteration within `produce()`

Reply:
The point is that this variable is now entirely redundant with setting the batch size to 0. in `produce()`, you can check the batch size directly from the client, or just get the request output, which will have size zero (in the old interface, output was not created if inference was skipped, but now a dummy, empty output is created).

Reply:
Fixed.